library/core/src/intrinsics/fallback.rs

```diff
@@ -110,3 +110,41 @@ impl const CarryingMulAdd for i128 {
         (low, high)
     }
 }
+
+#[const_trait]
+#[rustc_const_unstable(feature = "core_intrinsics_fallbacks", issue = "none")]
+pub trait DisjointBitOr: Copy + 'static {
+    /// See [`super::disjoint_bitor`]; we just need the trait indirection to handle
+    /// different types since calling intrinsics with generics doesn't work.
```

Comment:
What does "calling intrinsics with generics doesn't work" mean?

Also, can't the default be

```rust
#[inline]
unsafe fn disjoint_bitor<T>(a: T, b: T) -> T
where
    T: Default + Eq + BitOr + BitAnd
 {
    // Note that the assume here is required for UB detection in Miri!

    // SAFETY: our precondition is that there are no bits in common,
    // so this is just telling that to the backend.
    unsafe { super::assume(a & b == T::default()) };
    a | b
}
```